### PR TITLE
Feature/teacher lessons dashboard

### PR DESCRIPTION
**Pull request by Melizzo**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- Currently there is a generic teacher dashboard that does not have any additional state requested
## What is the new behavior?

- Upon loading the teacher dashboard, a get request for teacher obj, students and lessons array are requested and loaded into 
state.  Additionally, the information is now available for the lessons & students pages

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Screen Shot 2020-09-12 at 12 24 59 PM

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const setStudent= student => ({
 })
 
 export const setLessons= lessons => ({
-  type: 'SET_STUDENT',
+  type: 'SET_LESSONS',
   lessons
```
Comment (Melizzo) [inline src/actions/index.js:19]: This took me a hot second to figure out - but once I did, lessons were loaded immediately!

File: src/TeacherDashboard/TeacherDashboard.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+
+  useEffect (() => {
+    async function fetchData() {
+      await props.getTeacher()
+      await props.getStudents()
+      await props.getLessons()
+    } 
+    fetchData()
+    // eslint-disable-next-line 
+  }, [])
```
Comment (Melizzo) [inline src/TeacherDashboard/TeacherDashboard.js:19]: I'm not thrilled about the //eslint-disable right here - but I wanted to get this branch pushed up so ya'll could start on your pages.  The warning asked us to desctructor props, but I was having a few issues, I'll come back to this before final submission.

Comment (Billwilke42) [reply src/TeacherDashboard/TeacherDashboard.js:19]: I had to do this too, I'll try to read up more on this as well

File: src/TeacherDashboard/TeacherDashboard.js [hunk 3]
```diff
@@ -19,4 +34,21 @@ const TeacherDashboard = () => {
   )
 }
 
-export default TeacherDashboard;
\ No newline at end of file
+const mapStateToProps = ({ setStudents, setLessons, setTeacher }) => ({
```
Comment (Billwilke42) [inline src/TeacherDashboard/TeacherDashboard.js:37]: Nice work on this Redux, I know it can be tricky!

**Review by Billwilke42 — COMMENTED** [review]
Great work! I'll push mine up after we get this one merged and I deal with any conflict locally

**Review by Billwilke42 — APPROVED** [review]

**Review by KyleWong2510 — APPROVED** [review]
Great job!

**Pull request merged**